src/optim_problems/opt_cep.jl

```diff
@@ -161,45 +188,41 @@ function setup_opt_cep_generation_el!(cep::OptModelCEP,
                             opt_data::OptDataCEP)
     ## DATA ##
     set=cep.set
-    #nodes: nodes x region, infrastruct, capacity_of_different_tech...
-    nodes=opt_data.nodes
-    #cap_costs   tech x region, year, impact[USD, CO2]...
-    cap_costs=opt_data.cap_costs
-    #var_costs   tech x region, year, impact[USD, CO2]...
-    var_costs=opt_data.var_costs
-    #fix_costs   tech x region, year, impact[USD, CO2]...
-    fix_costs=opt_data.fix_costs
-    #techs       tech x [categ,sector,lifetime,effic,fuel,annuityfactor]
-    techs=opt_data.techs
+    #`costs::OptVariable`: costs[tech,node,year,account,impact] - annulized costs [USD in USD/MW_el, CO2 in kg-CO₂-eq./MW_el]`
+    costs = opt_data.costs
+    #`techs::OptVariable`: techs[tech] - OptDataCEPTech
+    techs = opt_data.techs
+    #`nodes::OptVariable`: nodes[tech,node] - OptDataCEPNode
+    nodes = opt_data.nodes
     #ts          Dict( tech-node ): t x k
     ts=ts_data.data
     #ts_weights: k - weight of each period:
     ts_weights=ts_data.weights
     #ts_deltas:  t x k - Δt of each segment x period
-    ts_deltas=1
+    ts_deltas=ts_data.deltas
 
 
     ## GENERATION ELECTRICITY ##
     # Calculate Variable Costs
-    push!(cep.info,"COST['var',impact,tech] = Σ_{t,k,node}GEN['el',t,k,node]⋅ ts_weights[k] ⋅ ts_deltas ⋅ var_costs[tech,impact] ∀ impact, tech_generation")
-    @constraint(cep.model, [impact=set["impact"], tech=set["tech_generation"]], cep.model[:COST]["var",impact,tech]==sum(cep.model[:GEN]["el",tech,t,k,node]*ts_weights[k]*ts_deltas*find_cost_in_df(var_costs,nodes,tech,node,impact) for node=set["nodes"], t=set["time_T"], k=set["time_K"]))
+    push!(cep.info,"COST['var',impact,tech] = Σ_{t,k,node}GEN['el',t,k,node]⋅ ts_weights[k] ⋅ ts_deltas[t,k]⋅ var_costs[tech,impact] ∀ impact, tech_generation")
+    @constraint(cep.model, [impact=set["impact"], tech=set["tech_generation"]], cep.model[:COST]["var",impact,tech]==sum(cep.model[:GEN]["el",tech,t,k,node]*ts_weights[k]*ts_deltas[t,k]*costs[tech,node,set["year"][1],"var",impact] for node=set["nodes"], t=set["time_T"], k=set["time_K"]))
     # Calculate Fixed Costs
-    push!(cep.info,"COST['cap_fix',impact,tech] = Σ_{node}CAP[tech,'new',node] ⋅ cap_costs[tech,impact] ∀ impact, tech_generation")
-    @constraint(cep.model, [impact=set["impact"], tech=set["tech_generation"]], cep.model[:COST]["cap_fix",impact,tech]==sum(cep.model[:CAP][tech,"new",node] *(find_cost_in_df(cap_costs,nodes,tech,node,impact)+find_cost_in_df(fix_costs,nodes,tech,node,impact)) for node=set["nodes"]))
+    push!(cep.info,"COST['cap_fix',impact,tech] = Σ_{t,k}(ts_weights ⋅ ts_deltas[t,k])/8760h ⋅ Σ_{node}CAP[tech,'new',node] ⋅ cap_costs[tech,impact] ∀ impact, tech_generation")
```

Comment:
if multiple years are run
Assumption: The years repeat in the same year (2016, 2016, 2016) for the cost data as they are only monte carlo

Reply:
What do you mean by monte carlo?

Reply:
The actual weather data of each year has the same statistical chance to occur in the modelled year